src/phpDocumentor/Descriptor/Namespace_.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.3
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor;
+
+use phpDocumentor\Reflection\Element;
+use phpDocumentor\Reflection\Fqsen;
+use phpDocumentor\Reflection\DocBlock;
+
+/**
+ * Represents a namespace and its children for a project.
+ */
+final class Namespace_ implements Element
+{
+    /**
+     * @var Fqsen Full Qualified Structural Element Name
+     */
+    private $fqsen;
+
+    /**
+     * @var DocBlock | Null
+     */
+    private $docBlock = null;
+
+    /** @var Namespace_[] $namespaces */
+    private $children = array();
+
+    /** @var Function_[] $functions */
+    private $functions = array();
+
+    /** @var Constant[] $constants */
+    private $constants = array();
+
+    /** @var Class_[] $classes */
+    private $classes = array();
+
+    /** @var Interface_[] $interfaces */
+    private $interfaces = array();
+
+    /** @var Trait_[] $traits */
+    private $traits = array();
+
+    /**
+     * Initializes the namespace.
+     *
+     * @param Fqsen $fqsen
+     * @param DocBlock $docBlock
+     */
+    public function __construct(Fqsen $fqsen, DocBlock $docBlock = null)
```

Comment:
A namespace cannot have a DocBlock unfortunately; it does not have a canonical declaration